### ISSUE TITLE
Adb exec-out not working properly

### ISSUE THREAD

**Issue by n0k0m3** [Egoistically/ALAuto#3]
`Adb exec-out screencap` is buggy for those who didn't have a proper Android SDK installation. Testing the command `adb exec-out screencap -p | sed s/\r\n/\n/` produces corrupted image.

I also tried many different of sed'ed and perl'ed flavor of this command and still gets corrupted file (tested on multiple machines with other users)

**Suggestion:** Offering a slower alternative for update_screen:
- Using native `adb shell screencap` to save a screenshot on emulator
- Use `adb pull` to pull the screenshot -> proper screenshot

Example:

adb.py, add to last:

```
    @staticmethod
    def pull(args):
        """Executes the command via adb shell

        Args:
            args (string): Command to execute.
        """
        cmd = ['adb', 'pull'] + args.split(' ')
        #Logger.log_debug(str(cmd))
        subprocess.call(cmd, shell=False, stderr=subprocess.DEVNULL)
```

utils.py, line 79, replace / add slower alternative method for screencap:

```
            Adb.shell("screencap /sdcard/screenshot.png")
            Adb.pull("/sdcard/screenshot.png")
            screen = cv2.imdecode(numpy.fromstring(open("screenshot.png","rb").read(),dtype=numpy.uint8),0)
```

While this method of updating screen will definitely be slower, the reliability is 100%

**Comment by n0k0m3** [comment]
Also the reason that I didn't make a pull request is that you may have different idea of how to implement this, so I'd leave to your own decision

**Comment by hjk22** [comment]
Sorry for replying to this thread, maybe I should have opened another issue, but I wanted to share my findings on this matter.
After reading the README.md, I was wondering why the `sed` command was needed. I looked through the code and found it was only used in the line in question (line 79, utils.py). The next question was why the output of the `screencap -p` command was pipelined to the `sed`, for what reason certain bytes needed to be deleted. I think it was added to address the [issue](https://blog.shvetsov.com/2013/02/grab-android-screenshot-to-computer-via.html) of the `adb shell` performing an EOL character conversion.
I looked through the code of [azurlane-auto](https://github.com/perryhuynh/azurlane-auto) and I found out that the author didn't change the output of the `screencap` command. I used his line and the script worked correctly. I added this line `cv2.imwrite("test.png", screen)` to the code in order to verify if the screenshot taken and then saved to the filesystem was corrupted, and it wasn't. 
After that I worked directly with `adb`, first using `exec-out` and then `shell`:
`adb -s SERIAL exec-out screencap -p > test2.png`
The screenshot taken (test2.png) wasn't corrupted.
`adb -s SERIAL shell screencap -p > test3.png`
This time test3.png **was** corrupted.
I have to add that I am using the latest Android SDK platform tools available ([29.0.4](https://developer.android.com/studio/releases/platform-tool)) and not the one suggested in the README.md file, and that I ran my tests on my Windows10 machine using Bluestacks4 as my Android emulator.
So, from my little research, I'm lead to believe that the `sed` "hack" was meant to resolve only the `shell` issue and that there wasn't any issue in the `exec-out`.
Of course I may be wrong, but if I am correct I think the best solution would be to restore this line to its previous version.
Changing this
`screen = cv2.imdecode(numpy.fromstring(Adb.exec_out(r"screencap -p | sed s/\r\n/\n/"),dtype=numpy.uint8),0)` 
to
`screen = cv2.imdecode(numpy.fromstring(Adb.exec_out('screencap -p'), dtype=numpy.uint8), 0)`.
Maybe it could also be a good idea to upgrade to the latest platform tools instead of using the r23 one, unless there is a specific reason not to of course.

**Comment by Egoistically** [comment]
This bot isn't specifically based on azurlane-auto but on one of its forks for the chinese server where sed and network settings were added. I remember I had a lot of trouble with exec-out and shell back in the day so as soon as I found a solution I just ignored whether or not it was the ideal one. 

~About the adb version, if I run the latest the images end up corrupted. Might be because of [this](https://i.imgur.com/pf6zMrN.png).
(I didn't bother testing all of the versions one by one, apparently r23 is for Android 6.0 and not 5.0 so what I just said doesn't make much sense but it works regardless).~

Thank you for your feedback, I will look into it.

Edit:
Looks like sed is only needed with old adb versions, when using the latest one it does indeed work fine without it.
